### DISCUSSION TITLE
ESP32 Becomes Unresponsive / Locks-Up / Drops WIFI / Needs Reboot

### DISCUSSION THREAD

**Discussion by Brummy** [tony-fav/tasmota-blerry#28] [Q&A] [answered]
For whatever reason, My ESP32 seems to stop working randomly. Sometimes once a day, sometimes multiple time, sometimes it'll go a few days.  I have to power cycle it to get it running again. 

Is this a Blerry issue?  A Tasmota issue?  A user issue!?

I'm using a NodeMCU-type ESP32 Devkit board.  Haven't had any similar problems with other uses of it with just Tasmota.  A buddy that also started using Blerry has had similar issues.  

I'm running 7 InkBird TH2 sensors right now (problem started when I only had one a couple of weeks ago).
Thanks,
Brummy

**Comment by tony-fav** [comment]
I don't have time for a full response to this at the moment but can you try `SetOption57 0` and report back if your average uptime improves? Also, jump in digiblur's discord for live help from a bunch of users and myself!

**Comment by Brummy** [comment]
Thanks, 
DigiBlur recommended trying the dev version of Tasmota Bluetooth. I switched to that. If it doesn't work, I'll try SO 57 0.
Will let you know either way. 
Appreciate the response. 
-Brummy

**Comment by tony-fav** [comment]
My longer thought is that it's asking a lot of the radio to 1) deal with the regular wifi connection, 2) deal with BLE passive and active scans, then 3) do a scan of available networks. You could run into power supply issues or just random issues that cause the radio to seize. I have not been able to reproduce the issue on a dev-board myself. So, your update will be greatly appreciated!

However, I did have an ESP32 plug (Wyze WLPP1CFH) that was showing similar behavior, uptimes of ~1 day. Turning SO57 off, and I hit 10 days before I restarted stuff for an unrelated reason.

**Comment by Brummy** [comment]
I've asked a buddy that's having the same issue try the SO57 option.  Will go a day or three and follow up (sooner if he or I see similar problems).

**Comment by osaether** [comment]
I have experienced this as well. I am using the TTGO T7 V1.5 ESP32 module and two Inkbird IBS-TH2 sensors.

**Comment by micahvandeusen** [comment]
I’m having the same problem. I have 4 Shelly 1 Plus with Tasmota32 solo1 BLE and thermostat either version 10.1.0.2 or 10.1.0.1. Most have only 1 sensor it is reading. They can go several days without issues and then require a reboot to start responding again. I’ll try the above option and report back if it’s not fixed.

**Comment by Brummy** [comment]
The dev version of Tasmota lasted about 40 hours before the problem popped back up. 

My buddy had it pop up with SO57 0 on the latest release version shortly after making that change. 

I'll try a different controller with only one inkbird on it. 

I'll also use a different flashing tool than Chrome, but don't think that really matters.

**Comment by tony-fav** [comment]
Since you are using a development board, were you able to capture serial log data to see what's happening? Like I said, I have been unable to reproduce the issue locally. If we are to propose to the Tasmota devs that there is an issue somewhere, we need log data.

**Comment by digiblur** [comment]
I'll set up a couple here on the latest dev build and dump them to a file.  I'll do some dev boards of various ones with the same files on various sensor types.

**Comment by devzwf** [comment]
I was not able to reproduce as well.
i am running both  dev branch and release branch at the moment 
with C3 and dev board 
both with : {"SetOption57":"ON"}

of course with blerry
let me know if i can help 

they all have to monitor :
about 5 to 7 PVVX
2 govee
2 Inkbird

**Comment by Brummy** [comment]
> 

Pardon my lack of experience posting in github and the format. 
Here's what I found from the time it last stopped working (2 log entries from today - is this what you need?):

#### First log entry found
Logger: homeassistant.components.mqtt
Source: components/mqtt/__init__.py:963
Integration: MQTT (documentation, issues)
First occurred: December 31, 2021, 4:57:18 AM (13 occurrences)
Last logged: 4:27:40 AM

Disconnected from MQTT server 192.168.1.216:1883 (7)


#### Second log entry found
Logger: paho.mqtt.client
Source: /usr/local/lib/python3.9/site-packages/paho/mqtt/client.py:2529
First occurred: December 31, 2021, 4:57:18 AM (13 occurrences)
Last logged: 4:27:40 AM

failed to receive on socket: [Errno 104] Connection reset by peer

**Comment by digiblur** [comment]
I think the log we are talking about is from Tasmota. I was going to drop the log via syslog to a file.

**Comment by devzwf** [comment]
@Brummy need to log from the device
the easiest is to push the log from taz to a syslog server and this way you have it available 

edit: whoops digi was faster than me :)

**Comment by Brummy** [comment]
Thanks @devzwf and @digiblur 
I've never used syslog, but just figured out how to set it up. I'll be ready next time it goes down.
Good thing about problems is it is an opportunity to learn new tools.

**Comment by hurricanepkt** [comment]
Just wanted to throw this in here. I've experienced the same issue with what i assume is the same device.  Would additional logs be of use or a distraction?

**Comment by micahvandeusen** [comment]
I’ve had this issue pop up again on one of mine after changing SetOption57. I set syslog to debug but didn’t get anything when it happened. I might have the log forwarding setup wrong.

**Comment by Brummy** [comment]
I don't really see anything in my Syslog either (not that I really know what I'm looking at!).

I set up Node Red to reset a WiFi outlet if the RSSI from the ESP32 is unavailable for 10 minutes.  It shuts off the outlet for one minute and then turn it back on again. From that I can narrow down where to look in Syslog (assuming that using RSSI "unavailable" is a valid way to do this.... log looks like its still getting readings from the sensors.

One thing I have noticed is that Bluetooth Active Scan comes back on automatically after a restart.  I don't know that this happening is problem in itself, but if it is actively scanning, is that creating more "work" for the ESP32 that may be causing this?

If I share logs here, what's the best way to do that?

**Comment by tony-fav** [comment]
Pastebin or something similar would be useful. As you are using a dev-board, I was thinking you would do serial logging, not syslogging. Serial log will see all the way through the crash where syslog will miss anything if the radio is dying somehow. 

The Inkbird sensors require active scanning to report their data. This isn't too much work for the ESP. While it's passively scanning, if it received a packet, it asks that device if it has anything more to say. If you're still worried, I'd point out that active scanning is enabled by default in ESPHome. https://esphome.io/components/esp32_ble_tracker.html

**Comment by Brummy** [comment]
Ah, gotcha. 
I'll look into the serial logging. 
Appreciate everyone's efforts to see if there's a solution (or at least understand what's happening)

**Comment by micahvandeusen** [comment]
It happened again with syslog configured. Nothing interesting it seems from what I can tell. For time reference, at 2022-01-06T03:24:01 it was marked as unavailable in home assistant and these are the last logs from the device.

[logs.txt](https://github.com/tony-fav/tasmota-blerry/files/7819438/logs.txt)

**Comment by tony-fav** [comment]
Thanks for the logs, as there's nothing reported useful over syslog, the error is likely the radio crashing for some reason. Serial logging at `seriallog 4` would be useful to see what's going on because that logging will persist after the radio is down.

**Comment by tony-fav** [comment]
I have uploaded some firmware binaries here https://github.com/tony-fav/tasmota-blerry/tree/dev-trusty-firmware/tested_firmware that I believe do not have the locking up problem discussed here. 
- `tasmota32-ble-berry.bin` for regular ESP32
- `tasmota32c3-ble-berry.bin` for ESP32C3
- `tasmota32solo1-linkind-ble-berry.bin` for Single Core ESP32 (and includes the Linkind Dimmer Module by pcdiem).

If anyone feels like testing these out and reporting back, we'd all appreciate it. After sufficient testing, I'll merge that branch into main. Thanks to flynmoose in digiblur's discord who was able to reproduce some of the crashing and we were able to identify a build that worked for him.

**Comment by gdavidp** [comment]
> Thanks for the logs, as there's nothing reported useful over syslog, the error is likely the radio crashing for some reason. Serial logging at `seriallog 4` would be useful to see what's going on because that logging will persist after the radio is down.

I put one of my ESP32Dev mini boards with rsyslog to a CentOS VM, with both Serial Log = 4 and Syslog = 4 Debug, will let you know when it crashes again. It usually takes a couple of days at most.

**Comment by Brummy** [comment]
Thanks @tony-fav. My test ESP32 running a single inkbird gave out after 40 hours.
I'll give this a try and follow up if it crashes or after a few days if not.

**Comment by hurricanepkt** [comment]
Was able to reproduce on what aliexpress calls "D1 mini ESP32 ESP-32 WiFi+Bluetooth Internet of Things development board based ESP8266 Fully functional"   

Logs are on a gist :

Transition from Working to error state : 
[https://gist.githubusercontent.com/hurricanepkt/b17a549d05c57310eb8edc5bc9900043/raw/affef19cf2c1b8c9f92bb47233759fbcc262b851/error_logs.txt](https://gist.githubusercontent.com/hurricanepkt/b17a549d05c57310eb8edc5bc9900043/raw/affef19cf2c1b8c9f92bb47233759fbcc262b851/error_logs.txt)

Hitting the reset button immediately works: 
[https://gist.githubusercontent.com/hurricanepkt/b17a549d05c57310eb8edc5bc9900043/raw/affef19cf2c1b8c9f92bb47233759fbcc262b851/error_state-restart_button-working_state.txt](https://gist.githubusercontent.com/hurricanepkt/b17a549d05c57310eb8edc5bc9900043/raw/affef19cf2c1b8c9f92bb47233759fbcc262b851/error_state-restart_button-working_state.txt)


Initial starting logs for reference : 

[https://gist.githubusercontent.com/hurricanepkt/b17a549d05c57310eb8edc5bc9900043/raw/affef19cf2c1b8c9f92bb47233759fbcc262b851/starting_logs.txt](https://gist.githubusercontent.com/hurricanepkt/b17a549d05c57310eb8edc5bc9900043/raw/affef19cf2c1b8c9f92bb47233759fbcc262b851/starting_logs.txt)

GreenwayDevices is a 2.4GHz only network for my iot devices. Shows up multiple times as i have multiple APs in the house.

**Comment by micahvandeusen** [comment]
Running tasmota32solo1-linkind-ble-berry.bin on a Shelly plus 1 I ran into the issue again. @tony-fav

**Comment by tony-fav** [comment]
If you could gather logs, describe your setup, flashing method, template, module, erase flash or no, etc. all of that would go a long way towards assisting. Also, please disable wifi rescan with `setoption57 0`

I'm pretty close to closing this issue and starting a new one with a description of all the requisite information to be helpful.

**Comment by tony-fav** [comment]
@hurricanepkt please try again with the firmware I've uploaded. I would also note that those boards in particular are known to have bad power supplies. Power availability can be a major issue when asking an ESP to do wifi and BLE concurrently

**Comment by micahvandeusen** [comment]
I am running a Shelly plus 1 in wall. OTA flash of tasmota32solo1-linkind-ble-berry.bin from Tasmota32 solo1 BLE and thermostat version 10.1.0.1 which was originally flashed by serial. It indeed had setoption57 0. The template/module is what is listed here https://templates.blakadder.com/shelly_plus_1.html. The syslogs had nothing interesting and this is all ready installed so not easy to get serial logs. I don't remember if I erased flash or not.

**Comment by micahvandeusen** [comment]
One thing I have noticed when it becomes unavailable on WiFi is the physical switch works but is very laggy and several seconds to a minute delayed.

**Comment by tony-fav** [comment]
@hurricanepkt looking at your logs I see the WiFi rescan shortly before crash. Can you try also disabling that feature (`SO57 0`)? A lot of these devices have only a small amount of extra power to use for other matters, especially when only powered by a computer's USB hub. That scan _could_ have started some instability in the radio due to power requirements.

**Comment by tony-fav** [comment]
After some discussion on discord and logs the following rule may help in case of wifi failures that just never reconnect

```
Rule3 
ON Wifi#Connected Do RuleTimer1 0 ENDON
ON Wifi#Disconnected Do RuleTimer1 60 ENDON
ON Rules#Timer=1 Do Restart 1 ENDON
```
Remember to activate with `Rule3 1`

This rule will restart device 60 seconds after a wifi disconnect if it has not reconnected.

Is anyone experiencing crashing so rapidly such a hacky workaround would not be useful? Like every few minutes?

**Comment by Brummy** [comment]
Just checked my test set up. It quit about 6:05pm (approx 7 hours ago). First time on new Tasmota bin file. 

Can copy the serial log from before and after if that’d be helpful (not sure you'd see anything you haven't already) 

If so, 5 minutes either side good? Can get to you later today if so. 

Will try the Rule3. When doing that, should I try SO57 0 at the same time? 

My full setup with 7 inkbirds on a different controller is using NodeRed to switch a WiFi outlet... I'm not logging, but looking at restart numbers it's doing it a several times a day. It waits for no WiFi signal for 10 minutes before power cycling.

Haven't looked at the history in HA, but I don't think it's happening at a frequency where Rule3's 60 seconds would be an issue. 

Hopefully Rule3 works... I really don't like physically restarting it via the outlet. 

Thanks

**Comment by tony-fav** [comment]
I would just throw the kitchen sink at it. SO57 0, restart rule, etc...

I'd really love to get everyone super stable without all of this, but there's so much variety in devices and so many things the Bluetooth functionality of the ESP32 depends on that it's kind of a waiting game to see if the SDK/Arduino Framework folks make it more stable.

**Comment by gdavidp** [comment]
The one I had running on the ESP32 mini board with all syslog turned on finally went offline. I can confirm that it did not report anything useful, just a Garbage Collection (GC) message and it was gone.

**Comment by mjoshd** [comment]
Mine is running on a MH-ET LIVE MiniKit (the ESP32 that fit's in the older QuinLED units). It also used to freeze up after a couple days and require a reboot. For what it's worth, I did a daily restart timer/rule combo and it's been pretty solid.

1. Set up 24hr timer to execute at midnight
```
Backlog Timers 1; Timer1 {"Enable":1,"Mode":0,"Time":"00:00","Window":0,"Days":"1111111","Repeat":1,"Action":3};
```

2. Add reboot to rule1 (also contains the original rule used to load blerry.be)
```
Backlog Rule1 ON System#Boot DO br load('blerry.be') ENDON ON Clock#Timer=1 do Restart 1 ENDON; Rule1 1;
```

**Comment by tony-fav** [comment]
Thanks @mjoshd that's maybe evidence that some error is building up over time rather than some random error causing a lock up. If it were random, I'd expect you to still see errors with that approach.

**Comment by davblair2000** [comment]
I have one setup on a nodemcu and the rule that Tony provided has kept me stable for 2 days now. Thanks!

I also have a Linkind and using the .bin provided.... again...rock solid! I'd recommend trying both if you can, they worked well for me.

Thanks for the work on the project and troubleshooting!

**Comment by jan-bee** [comment]
There was a problem in Tasmota FW, it looks like new fw (10.1.0 or 2022.01.2) solved problem in my case: https://github.com/arendst/Tasmota/issues/14262

**Comment by tony-fav** [comment]
@jan-bee yes, there was definitely a problem with memory leaks in a prior firmware due to an upstream error. I believe that is the error barbudor is referring too which would fix some issues if folks are on older firmware. The firmware I have posted are on a version which includes that fix as well.

Though Jason2866's comment that an Erase Flash may be required is an interesting one!

**Comment by tony-fav** [comment]
Just noticed the rule I posted here https://github.com/tony-fav/tasmota-blerry/issues/7#issuecomment-1008236720 was missing an `ON` so the restart would not actually trigger 🤦‍♂️

**Comment by gdavidp** [comment]
Looks like the the last Rule3 is solving my issue as a workaround. So far I have had one ESP32 restart on it's own and maintained WIFI connection. Thanks to Tony and all for this solution, I will keep watching them as it has been only 12 hours.

**Comment by tony-fav** [comment]
@Brummy how has your stability been?

**Comment by Brummy** [comment]
Sorry, Haven't had time to use the corrected rule 3 yet on test unit. 
Will update everything and see how it goes.

**Comment by tony-fav** [comment]
If anyone is willing to test, please try the latest -bluetooth builds available from 
https://github.com/tasmota/install/blob/main/firmware/development/tasmota32-bluetooth.bin
https://github.com/tasmota/install/blob/main/firmware/unofficial/tasmota32c3-bluetooth.bin
https://github.com/tasmota/install/blob/main/firmware/unofficial/tasmota32solo1-bluetooth.bin

There have been some bugfixes in the upstream NimBLE library that are now in Tasmota and the EQ3 TRV driver is now disabled by default which may have caused some problems for people. Note, these aren't any changes in BLErry but changes in Tasmota and its upstream libraries. I have yet to find any place where the BLErry code could be causing these issues directly.

If you continue to have issues, please indicate what ESP based device you are using. ESP32's that look like D1 minis are especially subject to issues as they are known to have bad voltage regulators that do not supply enough current to the ESP32.

**Comment by Brummy** [comment]
Trying it. Crashed shortly after I set it up but that may have been the controller (reset to default Tasmota AP for setting up) 
Just redid it, so we'll see. Can't serial log right now but will get it hooked up shortly.

**Comment by micahvandeusen** [comment]
I’m also trying it currently on several devices and no issues so far (though it went days without issues on the old version). The daily scheduled restart has kept them working so far. I disabled that to see if this fixes the root issue.

**Comment by tony-fav** [comment]
> Trying it. Crashed shortly after I set it up but that may have been the controller (reset to default Tasmota AP for setting up) Just redid it, so we'll see. Can't serial log right now but will get it hooked up shortly.

If you can, do an erase flash then flash latest dev. Jason2866 mentioned on a Tasmota issue that somehow some things that were screwing with BLE stability on old frameworks didn't get taken care of by an OTA and `Reset 1` but did with a full flash erase.

**Comment by gdavidp** [comment]
I threw on this version on to one of my ESP32s with the standard Rule1 and Blerry:

```
Program Version	2022.01.3(bluetooth)
Build Date & Time	2022-01-19T22:07:21
Core/SDK Version	2_0_2/v4.4-2.0.2
```

I did erase the flash so it should have been a blank slate, will update if goes down.

**Comment by bschatzow** [comment]
Good timing, mine crashed last night after 48 hours.  Trying it now.

**Comment by micahvandeusen** [comment]
Had a Shelly plus 1 crash with the new firmware. However I didn’t clear flash.

**Comment by gdavidp** [comment]
Newest Tasmota DEV firmware does not do anything for me, it is in this usual loop, displaying this every second:

15:21:05.282 WIF: Connect failed as AP cannot be reached

**Comment by osaether** [comment]
I erased the device and flashed the new FW above today and it became unresponsive after about one hour.

**Comment by BFLK** [comment]
I am using a so called ESP-32 D1 mini as a MQTT gateway for some Xiaomi BLE sensors and I think I am facing the same issue after upgrading Tasmota from 9.5.0 to 10.1.0. The device is unresponsive (MQTT and HTTP) and the issue is solved after a power reset - however it reoccurs after 1-2 days. I will rollback to 9.5.0 but it would be good if someone will be able to understand and mitigate this issue.

**Reply by tony-fav** [reply]
Are you actually using Blerry or the source level Tasmota driver for the Xiaomi sensors?

**Reply by BFLK** [reply]
I am using tasmota32-bluetooth.bin (the pre-compiled one) and for the Xiaomi Temperature and Humidity sensors I replaced the original firmware with the ATC one. There was no configuration needed on Tasmota side besides of the MQTT settings and enabling BLE. If there's a way to understand if I am using Blerry or the source level Tasmota driver, I'd be happy to check.

When checking the information section again, I realized that the press of the reset button is not the same as the power reset: Restart Reason | Software reset CPU

**Reply by tony-fav** [reply]
The repository you are in is for a project which uses Tasmota but is not Tasmota. If you are using stock Tasmota but not the project in this repository, please try to open a Tasmota issue on the Tasmota repository https://github.com/arendst/Tasmota/issues/new?assignees=&labels=&template=Bug_report.md

**Reply by BFLK** [reply]
I just realized the error. I was just googling around in order to understand this issue and I reached this topic. One way or another I managed to miss the fact that it was not in the Tasmota repository. Sorry for loosing your time and thank a lot for the fast replies. Have a nice evening!

**Reply by tony-fav** [reply]
It's actually great (for us) that you are getting the error *NOT* using this repo! Kind of another reason to believe that it's not this projects fault ;-) Thanks for doing the issue report if you get to it!

**Reply by BFLK** [reply]
I've raised an issue: https://github.com/arendst/Tasmota/issues/14583

**Comment by micahvandeusen** [comment]
So was it stable with 9.5.0? That might mean it was something in a recent version?

**Comment by BFLK** [comment]
Yes, it was stable for a long time 9.5.0 as I don't recall ever resetting the device. Something to add, after reading this topic, I just pressed the reset button on the ESP-32 D1 mini and it recovered - I don't know if this is in any way different from a power reset. Something that I didn't state in the initial post is that I'm using the Tasmota32-bluetooth build.

**Comment by micahvandeusen** [comment]
The nightly reboot timer worked as a mitigation for the unresponsive/wifi dropping issue on my Shelly plus 1s. That is until just recently I had one of them go offline.

I tried the latest dev version of Tasmota (11.0.0.4) doing an OTA upgrade from https://github.com/tasmota/install/blob/main/firmware/unofficial/tasmota32solo1-bluetooth.bin?raw=true and I think it fixed the problem as I haven’t had any issues so far without the reboot timer.

**Reply by Brummy** [reply]
I've not replied lately. Kinda gave up and just used a smart plug with a Node Red automation to address the problem. Works good enough. 
A couple of weeks ago I used one of the ESP32 I had trouble with before to set up two more InkBirds for my garden seedling nursery. Used the new version of Blerry and it's been working great. 
I'm not logging it, but I do see the boot count increasing, but due to software resets. It appears the new version overcomes possible Tasmota (11.0.0.3) challenges. 
Thanks again Tony!

**Comment by iLeftheris** [comment]
After firmware upgrade to **11.0.0.4(bluetooth)** my **ESP32-D0WDQ6** is working continuously for 48 hours
2 ATCpvvx devices with **SetOption57 1**

**Comment by patrickdk77** [comment]
Same here on 11.0.0.3 no reboots since it upgraded 15days ago
11.0.0.1 though still only 5days max I see

**Reply by BFLK** [reply]
The issue is not reproducible anymore in Tasmota v11.0.0 Olivia, so it was for sure something from upstream, and not tasmota-blerry. I now have 9days and 16hours of uptime on my ESP-32 D1 mini.

**Comment by tony-fav** [comment]
Really glad to hear the latest Tasmota is stable without custom compiles! I am also working on a compatibility layer to use the new BLE driver that's being developed, and it is working great!

**Comment by ceinmart** [comment]
Hi!

I'm quite new to using ESP devices... 
Trying to do my BLE Tracker using the Tasmota instead ESPHome , since Tasmota appears to be more "friendly"... 

So, I have a ESP32-DevKit (ESP32-wroom-32) , with last Tasmota version **13.2.0**(bluetooth). 
The berry appears to be installed successfully, at least after running the script to return the message that was installed. 

However, I got into the issue where my ESP began to restart without stop, 2-3 seconds after boot, restarted again. 
I set it to 4 level debug to serial and got this :  **Guru Meditation Error: Core  1 panic'ed (IntegerDivideByZero). Exception was unhandled.
**
Have no BLE device configured yet... could this be the reason? or some code flaws?
Is there a way to enable a code debug and dump at serial, to show where they are trying to divide by zero?


```
3:10:55.347 CFG: Saved, Count 395, Bytes 4096
13:10:58.918 MQT: tasmota/discovery/xxxxxxx/config = {"ip":"xxxxxx","dn":"esp32_ble_tracker1","fn":["esp32_ble_tracker1",null,null,null,null,null,null,null],"hn":"esp32-ble-tracker1-1840","mac":"xxxxxxx","md":"ESP32-DevKit","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"13.2.0","t":"esp32_ble_tracker1","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"bat":0,"dslp":0,"sho":[],"sht":[],"ver":1} (retained)
13:10:59.010 MQT: tasmota/discovery/xxxxxxx/sensors = {"sn":{"Time":"2023-11-03T13:10:59"},"ver":1} (retained)
Guru Meditation Error: Core  1 panic'ed (IntegerDivideByZero). Exception was unhandled.

Core  1 register dump:
PC      : 0x400eebbc  PS      : 0x00060c30  A0      : 0x800f061e  A1      : 0x3ffd3de0  
A2      : 0x00000001  A3      : 0x3ffd5ba0  A4      : 0x00000000  A5      : 0x00000018  
A6      : 0x00000001  A7      : 0x00000001  A8      : 0x3ffd5bb8  A9      : 0x00000001  
A10     : 0x3ffd0c8c  A11     : 0x3ffd5bd0  A12     : 0x0000aec5  A13     : 0x3ffd5bc0  
A14     : 0x00000008  A15     : 0x3ffd3dd0  SAR     : 0x0000000a  EXCCAUSE: 0x00000006  
EXCVADDR: 0x00000000  LBEG    : 0x40095eb1  LEND    : 0x40095ec1  LCOUNT  : 0xffffffff  


Backtrace: 0x400eebb9:0x3ffd3de0 0x400f061b:0x3ffd3ea0 0x400f1abd:0x3ffd3ec0 0x400f1e2e:0x3ffd3ee0 0x400f2b43:0x3ffd3f60 0x4012834c:0x3ffd3fa0 0x4012aec1:0x3ffd3fd0 0x4011798d:0x3ffd3ff0 0x401525a2:0x3ffd4010 0x4015f14e:0x3ffd4030 0x400e57af:0x3ffd40a0




ELF file SHA256: 0000000000000000

Rebooting...

```

**Reply by devzwf** [reply, marked as answer]
Please try with tasmota v11.x train 
and not tasmota v13.x.x for your sanity

**Reply by ceinmart** [reply]
Thanks @devzwf  , I did the downgrade, and is running for couple hours without issues. 
This version also shows the ESP temperature, which tasmota v13 didn't... 
Appears to run smoothly ...

**Comment by Qowy** [comment]
So what is the current status here, are we still supposed to use tasmota 11?

I tried with 14.5 and had interesting results.
Both a Shelly Gen3 (with esp32-c3) and a NodeMCU with a ESP32-D0WD-V3 are totally unusable.

They lock up (or probably just the wifi becoming unresponsive) soon after enabling BLE in the settings. And executing blerry.be with a singe device in config basically kills them immediately.

Ironically I had the best luck with shelly plus 1 with a ESP32-U4WDH-S and using tasmota32solo1 That at least runs blerry and so far has not locked up for a few hours.

So is there a list/experience with known good espChip/tasmotaVersion combinations?

**Reply by tony-fav** [reply]
I think most of us (even me) have moved on to Bluetooth Proxy through ESPHome. The stability issues on Tasmota really made this project volatile.

**Reply by Qowy** [reply]
Ah ok, I am currently using a Shelly script with the native shelly firmware and that also works fine for bridging to MQTT.
But I have some existing tasmota devices that i wanted to use as a secondary receiver.

I am not using HomeAssistant so i need a more General solution. And I cannot rely on the HomeAssistant Integrations to decode the data.

I will then probably use the Berry code and Drivers as an example to write my own script for some simple sensors.
